### PR TITLE
Support for auxiliary inputs

### PR DESCRIPTION
**Pull request by samgelman**
Adding new feature to allow auxiliary inputs to be passed to network modules as keyword arguments in `forward()`. 

- Auxiliary data is placed in the main `.tsv` file for the dataset. The typical columns in that file are `variant`, `num_mutations`, and `score`. You can add columns for `aux1`, `aux2`, etc.. In practice, the column names should probably be more descriptive, for example something like `parent_id` instead of `aux1`. Multiple aux columns are supported. Integers, floats, and strings are supported, but a single column should not mix datatypes. 
- There are new arguments in `train_target_model.py` which allow you to specify auxiliary inputs. The main argument is `--aux_input_names`. This is where you can specify the names of the auxiliary data columns. So you would call the script with `—aux_input_names aux1 aux2` to specify that the information in columns `aux1` and `aux2` should be passed in to the model layers. The other argument is `--aux-formats` which allows you to specify the data type of each aux column, either `tensor`, `numpy`, or `string`. By default, numerical columns are fed through the network as PyTorch tensors. This is what you want if you’re using the auxiliary information in calls to PyTorch functions like for computing loss. You have the option to specify ‘numpy’ if you want a particular column fed through the network as a NumPy array. You may want this if you’re using the auxiliary data in Python control flow or on the CPU side of the processing rather than the GPU side.
- You need to set up your network architecture to accept the auxiliary inputs. There are 2 components to this. First, your model layers need to be wrapped in my custom `SequentialWithArgs` module instead of the PyTorch default `nn.Sequential` module. My `SequentialWithArgs` module, which is a subclass of `nn.Sequential`, has custom logic to accept in and pass along keyword arguments to any model layers which accept keyword arguments. This is how the auxiliary information is passed along to model layers. The second component to the network architecture is to make sure that your custom layer, which needs the auxiliary information, accepts keyword arguments in its forward function. 
- If you’re modifying my `AttnModel` or `TransferModel` classes, they are already wrapped in `SequentialWithArgs`. So all you’d have to do is add your new layer, which accepts `kwargs` in its `forward()` function, and your new layer should be able to access the auxiliary information automatically. 
- Auxiliary inputs are only supported for experimental data / finetuning, not for pretraining.

### CODE REVIEW

**Comment by agitter** [comment]
Should the new auxiliary support be documented in the readme somewhere?

**Comment by brycejoh16** [comment]
I reviewed the files and it looks correct to me. The only thing that might help is if @samgelman  shares the exact regression args file and tsv dms file he used to do some of the unit tests, so I can get that up and running when I am starting up the runs.

**Comment by samgelman** [comment]
> Should the new auxiliary support be documented in the readme somewhere?

The relevant arguments show up if you call the script with the `-h` flag, and because this isn't a core functionality for what we did in the paper, I don't think we need to feature it in the README
